CodeAssignment_MVVM_Combine/Sources/Controllers/DetailViewController.swift

```diff
@@ -0,0 +1,309 @@
+        let stackView = UIStackView(arrangedSubviews: [watchersCountLabel, forksCountLabel, openIssuesCountLabel])
+        stackView.axis = .vertical
+        // subView間のspacing設定
+        stackView.spacing = 12
+        stackView.alignment = .trailing
+        return stackView
+    }()
+    /// backgroundViewに入れるSubViewを持つStackView
+    private lazy var mainStackView: UIStackView = {
+        let stackView = UIStackView(arrangedSubviews: [
+            userImageContentView, userNameLabel, repositoryNameLabel, descriptionLabel, starLanguageStackView, countStackView
+        ])
+        stackView.axis = .vertical
+        stackView.spacing = 20
+        return stackView
+    }()
+    
+    init(repository: Repositories.Repository) {
+        super.init(nibName: nil, bundle: nil)
+        self.configure(with: repository)
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupUI()
+    }
+}
+
+extension DetailViewController {
+    private func configure(with model: Repositories.Repository) {
+        let defaultImage = UIImage(systemName: "person.circle")?.withTintColor(.systemMint, renderingMode: .alwaysOriginal)
+        
+        repositoryNameLabel.text = model.name
+        descriptionLabel.text = model.description
+        userNameLabel.text = model.owner.userName
+        languageNameLabel.text = model.language
+        // UIButton.Configurationのtitleを更新
+        if var config = starButton.configuration {
+            config.attributedTitle = AttributedString(
+                "\(formatNumberToStringWithSeparator(model.stargazersCount)) stars",
+                attributes: AttributeContainer([
+                    .font: UIFont.systemFont(ofSize: 20, weight: .regular),
+                            .foregroundColor: UIColor.black.withAlphaComponent(0.8)
+                ]))
+            starButton.configuration = config
+        }
+        
+        watchersCountLabel.text = "\(formatNumberToStringWithSeparator(model.watchersCount)) watchers"
+        forksCountLabel.text = "\(formatNumberToStringWithSeparator(model.forksCount)) forks"
+        openIssuesCountLabel.text = "\(formatNumberToStringWithSeparator(model.openIssuesCount)) issues"
+        
+        if let url = URL(string: model.owner.profileImageString) {
+            userImageView.sd_setImage(with: url, placeholderImage: defaultImage) { [weak self] (image, error, _, _) in
+                guard let self else { return }
+                if let error {
+                    // ロード中にエラーが発生する場合や、URLが無効な場合はdefaultの画像を表示
+                    print("error: \(error.localizedDescription)")
+                    self.userImageView.image = defaultImage
+                } else {
+                    self.userImageView.image = image
+                }
+            }
+        } else {
+            userImageView.image = defaultImage
+        }
```

Comment:
guardで書いてあげたほうが入れ子にならなくて読みやすいかな。

```suggestion
        guard let url = URL(string: model.owner.profileImageString) else { 
            userImageView.image = defaultImage
            return
        }
        userImageView.sd_setImage(with: url, placeholderImage: defaultImage) { [weak self] (image, error, _, _) in
            guard let self else { return }
            guard let error else {
                self.userImageView.image = image
                return
            }
            self.userImageView.image = defaultImage
        }
```

Reply:
[c984c15]にてコード修正しました！ご確認お願いします！